### PR TITLE
[snmp] Allow system with no ports in config db run without errors

### PR DESCRIPTION
**Pull request by liorghub**
**What I did**
Allow system with no ports in config db run without errors.
This is needed for modular system which should boot properly without line cards.

**How I did it**
Remove snmpagent error exit if there are no ports in config DB or in counters DB.

**How to verify it**
Run snmpwalk on the root oid.

**Important**
This PR must be merged only after PR https://github.com/Azure/sonic-py-swsssdk/pull/109 is merged.

### CODE REVIEW

**Comment by liorghub** [comment]
> * Wrong name for an argument in a c

Function get_interface_oid_map was modified to accept 2 arguments (PR https://github.com/Azure/sonic-py-swsssdk/pull/109).

**Comment by liorghub** [comment]
Important
This PR should be merged only after PR Azure/sonic-py-swsssdk#109 is merged.

**Comment by liorghub** [comment]
@qiluo-msft checkers are failing from the same reason I stated above.

It is failing since in this commit we call function get_interface_oid_map in its new form.
The signature of get_interface_oid_map was modified by me (hence the failure) in PR Azure/sonic-py-swsssdk#109.
Therefore PR Azure/sonic-py-swsssdk#109 must be merged before we can rerun the checkers for this PR.

File: tests/test_mibs.py [hunk 1]
```diff
@@ -30,3 +35,16 @@ def test_init_sync_d_lag_tables(self):
 
         self.assertTrue("PortChannel_Temp" in lag_name_if_name_map)
         self.assertTrue(lag_name_if_name_map["PortChannel_Temp"] == [])
+
+    @mock.patch('swsssdk.dbconnector.SonicV2Connector.get_all', mock.MagicMock(return_value=({})))
```
Comment (qiluo-msft) [inline tests/test_mibs.py:39]: >swsssdk [](http://example.com/codeflow?start=17&length=7)

swsssdk or swsscommon?

Comment (qiluo-msft) [reply tests/test_mibs.py:39]: Did you run `python3 -m pytest -s -v -k test_init_sync_d_interface_tables` ?

**Review by qiluo-msft — DISMISSED** [review]

File: tests/test_mibs.py (hunk 1 above)
Comment (liorghub) [reply tests/test_mibs.py:39]: swsssdk is the right one (method get_all is part of class SonicV2Connector which is implemented in sonic-buildimage/sonic-py-swsssdk/src/swsssdk/dbconnector.py).
Yes, I ran python3 -m pytest -s -v -k test_init_sync_d_interface_tables.

**Comment by qiluo-msft** [comment]
Could you solve the conflicts?

**Comment by qiluo-msft** [comment]
This is the conflict github reported:
image

**Comment by liorghub** [comment]
> This is the conflict github reported:
> image

@qiluo-msft I resolved the conflict, thanks for the heads up.

**Comment by liorghub** [comment]
@qiluo-msft 
Tests are failing due to the following issue:
> if_name_map_util, if_id_map_util = port_util.get_interface_oid_map(db_conn, blocking=False)
> TypeError: get_interface_oid_map() got an unexpected keyword argument 'blocking'

PR https://github.com/Azure/sonic-py-swsssdk/pull/109 contains the required change in get_interface_oid_map() and it is merged, however tests ran with the old version of the function, do you know why?

**Review by qiluo-msft — APPROVED** [review]

**Comment by qiluo-msft** [comment]
> PR [Azure/sonic-py-swsssdk#109](https://github.com/Azure/sonic-py-swsssdk/pull/109) contains the required change in get_interface_oid_map() and it is merged, however tests ran with the old version of the function, do you know why?

You need to move submodule in sonic-buildimage repo.

**Comment by liorghub** [comment]
> > PR [Azure/sonic-py-swsssdk#109](https://github.com/Azure/sonic-py-swsssdk/pull/109) contains the required change in get_interface_oid_map() and it is merged, however tests ran with the old version of the function, do you know why?
> 
> You need to move submodule in sonic-buildimage repo.

@qiluo-msft 
Thanks for the clarification.
Submodule update PR - https://github.com/Azure/sonic-buildimage/pull/8696

**Comment by liorghub** [comment]
@liat-grozovik 
We should rerun checkers only after I will update the hash of sonic-py-swsssdk submodule in sonic-buildimage.
I will update the hash right after https://github.com/Azure/sonic-py-swsssdk/pull/114 is merged.

**Comment by qiluo-msft** [comment]
Please update the hash of sonic-py-swsssdk submodule in sonic-buildimage.

**Comment by liorghub** [comment]
@qiluo-msft sonic-buildimage master is updated with the latest sonic-py-swsssdk hash (see https://github.com/Azure/sonic-buildimage/pull/8757) can you please rerun this PR?

**Comment by qiluo-msft** [comment]
PR owner could retrigger checkers by "/azpw run"

**Pull request merged**